sdk/appconfiguration/app-configuration/types/app-configuration.d.ts

```diff
@@ -0,0 +1,387 @@
+     * @param id The id of the configuration setting to set to read-only.
+     */
+    setReadOnly(id: ConfigurationSettingId, options?: SetReadOnlyOptions): Promise<SetReadOnlyResponse>;
+    /**
+     * Makes the key's value writable again
+     * @param id The id of the configuration setting to make writable.
+     */
+    clearReadOnly(id: ConfigurationSettingId, options?: ClearReadOnlyOptions): Promise<ClearReadOnlyResponse>;
+}
+
+/**
+ * Options for clearReadOnly
+ */
+export declare interface ClearReadOnlyOptions extends HttpOnlyIfUnchangedField, RequestOptionsBase {
+}
+
+/**
+ * Response when clearing the read-only status from a value
+ */
+export declare interface ClearReadOnlyResponse extends ConfigurationSetting, SyncTokenHeaderField, HttpResponseField<SyncTokenHeaderField> {
+}
+
+/**
+ * Configuration setting with extra metadata from the server, indicating
+ * its etag, whether it is currently readOnly and when it was last modified.
+ */
+export declare interface ConfigurationSetting extends ConfigurationSettingParam {
+    /**
+     * Whether or not the setting is read-only
+     */
+    readOnly: boolean;
+    /**
+     * The date when this setting was last modified
+     */
+    lastModified?: Date;
+}
+
+/**
+ * Fields that uniquely identify a configuration setting
+ */
+export declare interface ConfigurationSettingId {
+    /**
+     * The key for this setting
+     */
+    key: string;
+    /**
+     * The label for this setting. Leaving this undefined means this
+     * setting does not have a label.
+     */
+    label?: string;
+    /**
+     * The etag for this setting
+     */
+    etag?: string;
+}
+
+/**
+ * Necessary fields for updating or creating a new configuration setting
+ */
+export declare interface ConfigurationSettingParam extends ConfigurationSettingId {
+    /**
+     * The content type of the setting's value
+     */
+    contentType?: string;
+    /**
+     * The setting's value
+     */
+    value?: string;
+    /**
+     * Tags for this key
+     */
+    tags?: {
+        [propertyName: string]: string;
+    };
+}
+
+/**
+ * Standard base response for getting, deleting or updating a configuration setting
+ */
+export declare type ConfigurationSettingResponse<HeadersT> = ConfigurationSetting & HttpResponseField<HeadersT> & Pick<HeadersT, Exclude<keyof HeadersT, "eTag">>;
+
+/**
+ * Options for deleting a ConfigurationSetting.
+ */
+export declare interface DeleteConfigurationSettingOptions extends HttpOnlyIfUnchangedField, RequestOptionsBase {
+}
+
+/**
+ * Response from deleting a ConfigurationSetting.
+ */
+export declare interface DeleteConfigurationSettingResponse extends SyncTokenHeaderField, HttpResponseFields, HttpResponseField<SyncTokenHeaderField> {
+}
+
+/**
+ * Headers from getting a ConfigurationSetting.
+ */
+export declare interface GetConfigurationHeaders extends SyncTokenHeaderField {
+    /**
+     * A UTC datetime that specifies the last time the resource was modified.
```

Comment:
I'm guessing this is formatted as rfc7232? An example might be nice, e.g. `Tue, 15 Nov 1994 12:45:26 GMT`

Reply:
Will parse it into a date.

Reply:
Remove header suffix as well.